src/common/AndroidEnvironmentSetup.ts

```diff
@@ -101,12 +101,14 @@ export class AndroidEnvironmentSetup extends BaseSetup {
 
     public async isAndroidSdkRootSet(): Promise<string> {
         return new Promise<string>((resolve, reject) => {
-            if (AndroidSDKUtils.isAndroidSdkRootSet()) {
+            const root = AndroidSDKUtils.getAndroidSdkRoot();
+            if (AndroidSDKUtils.isAndroidSdkRootSet() && root !== undefined) {
```

Comment:
Given that `AndroidSDKUtils.getAndroidSdkRoot()` now more or less definitively ascertains the presence of the Android SDK, can we do away with the `AndroidSDKUtils.isAndroidSdkRootSet()` method altogether, in favor of the pattern:
```javascript
if (AndroidSDKUtils.getAndroidSdkRoot()) {
    // Android SDK is valid, do stuff.
}
```
It feels more and more unnecessary and burdensome to maintain two methods for such a similar purpose.

Reply:
An alternative would be to just take it out of this line, in favor of the sole `if (root)` condition. But its existence in general is starting to feel superfluous to me.